### PR TITLE
Handle symlinks directly

### LINKED ISSUE #323: wal-e backup-fetch hangs after several segments - s3
I'm trying to restore a backup with backup-fetch using an S3 compatible store,  Wal-e lists out several segments and then just sits there indefinitely. 
Wal-e version 1.0.3
Postgres 9.4
Don't really have any other output at this time.

### PR DESCRIPTION
**Pull request by cyberdelia**
Create symlinks directly, rather than letting the tar module do it
for us, to workaround this python issue: https://bugs.python.org/issue12800

This should fix #323.

### CODE REVIEW

File: wal_e/tar_partition.py [hunk 1]
```diff
@@ -281,17 +281,13 @@ def tarfile_extract(fileobj, dest_path):
...
-                tar.extract(member, path=dest_path)
-
             if member.issym():
-                # It does not appear possible to fsync a symlink, or
-                # so it seems, as there is no portable way to open()
-                # one to get a fd to run fsync on.
-                pass
+                os.symlink(member.linkname, os.path.join(dest_path, member.name))
             else:
+                if member.isreg() and member.size >= pipebuf.PIPE_BUF_BYTES:
```
Comment (fdr) [inline wal_e/tar_partition.py]: Why was this code restructured? It adds more code within branches when it seems no branch is necessary.

File: wal_e/tar_partition.py [hunk 2]
```diff
@@ -281,17 +281,13 @@ def tarfile_extract(fileobj, dest_path):
...
-                cat_extract(tar, member, relpath)
-            else:
-                tar.extract(member, path=dest_path)
-
             if member.issym():
-                # It does not appear possible to fsync a symlink, or
-                # so it seems, as there is no portable way to open()
-                # one to get a fd to run fsync on.
-                pass
+                os.symlink(member.linkname, os.path.join(dest_path, member.name))
```
Comment (fdr) [inline wal_e/tar_partition.py]: Should probably include a reference to the bug in Python.

**Comment by fdr** [comment]
Thanks for digging this one up.

**Comment by fdr** [comment]
If you squash and fix the build failure (which is from formatting), I'll commit it.

**Comment by cyberdelia** [comment]
Done.

**Pull request merged**